OpenRA.Game/Sound/SoundDevice.cs

```diff
@@ -26,7 +26,7 @@ public interface ISoundEngine : IDisposable
 		void SetAllSoundsPaused(bool paused);
 		void StopAllSounds();
 		void SetListenerPosition(WPos position);
-		void SetSoundVolume(float volume, ISound music, ISound video);
+		void SetSoundVolume(float volume, ISound sound);
```

Comment:
I don't exactly get how this works in either implementation - everywhere this method is used we apply `InternalSoundVolume` property onto `music` and `video `objects. But InternalSoundVolume is derived from SoundVolume which in term is `Game.Settings.Sound.SoundVolume` setting:
https://github.com/OpenRA/OpenRA/blob/5d8c9a560aba9feb88be9995365aa78b90487761/OpenRA.Game/Sound/Sound.cs#L281-L294

So if I'm correct, previously set MusicVolume and VideoVolume settings get overriden.

It is also applied in other various places like
https://github.com/OpenRA/OpenRA/blob/5d8c9a560aba9feb88be9995365aa78b90487761/OpenRA.Game/Sound/Sound.cs#L150-L155

Reply:
ok, I got how SetSoundVolume worked - `music` and `video` are passed there so that they can be excluded from updating the sound volume. So `SetSoundVolume` applies new value to everything that is not music and not video.

The implementation here however does exactly the opposite - applies the value to the object passed as parameter, which is not the expected behaviour on `SoundVolume` and `SoundVolumeModifier`.